frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -245,3 +266,80 @@ export const removeResourceConnection = (
     return Promise.all(patches);
   });
 };
+
+export const cleanUpWorkload = (
+  resource: K8sResourceKind,
+  workload: TopologyDataObject,
+): Promise<K8sResourceKind[]> => {
+  // cleanup all the items for a workoad.
+  const reqs = [];
+  const webhooks = [];
+  let webhooksAvailable = true;
+  const deleteModels = [BuildConfigModel, DeploymentConfigModel, ServiceModel, RouteModel];
+  const knativeDeleteModels = [
+    ConfigurationModel,
+    RevisionModel,
+    KnativeServiceModel,
+    KnativeRouteModel,
+    BuildConfigModel,
+    ImageStreamModel,
+  ];
+  const deploymentsAnnotations = _.get(resource, 'metadata.annotations', {});
+  const isKnativeResource = _.get(workload, 'data.isKnativeResource', false);
+  const gitType = detectGitType(deploymentsAnnotations['app.openshift.io/vcs-uri']);
+  const resourceData = _.cloneDeep(resource);
+  const batchDeleteRequests = (models, resourceObj, strict = false) => {
+    models.map((model) => {
+      if (strict) {
+        const resourceFound = !!_.find(workload.resources, (res: K8sResourceKind) => {
+          return res && res.kind === model.kind && res.metadata.name === resource.metadata.name;
+        });
+        return resourceFound ? reqs.push(k8sKill(model, resource)) : null;
+      }
+      return reqs.push(k8sKill(model, resourceObj));
+    });
+  };
+
+  if (isKnativeResource) {
+    // delete knative resources
+    const knativeRoute = _.find(workload.resources.ksroutes, { kind: 'Route' });
```

Comment:
it would be good to rely on knative services instead of routes if `metadata.name` is as expected

Reply:
@invincibleJai  Services are having generated names, looks like the only object to rely on to get the original name is `ksroutes` object.

Reply:
@karthikjeeyar got the issue, knative services are not available in resources. what you saw would be k8s resources.

Reply:
https://github.com/karthikjeeyar/console/pull/6